File: LAB1/laboratorio1_poo.py
# /**
#  *  Laboratorio 1 POO python
#  *  Por: Jerónimo Hoyos Botero
# */
import random

while True:
    tamano = int(input("Ingrese tamaño de lista menor o igual a 12: "))
    if 0 < tamano and tamano <= 12:
        break
A = []
B = []

for i in range(tamano):
    valor = int(input("Ingrese valor entero para el indice " + str(i) + " de A: "))
    A.append(valor)

for i in range(tamano):
    valor = random.randint(1,20)
    B.append(valor)

suma_AyB = []

for i in range(tamano):
    valor = A[i] + B[i]
    suma_AyB.append(valor)

producto_AyB = []

for i in range(tamano):
    valor = A[i] * B[i]
    producto_AyB.append(valor)

pares_enterosA = 0

for i in range(tamano):

    if A[i]%2 == 0:
        pares_enterosA+=A[i]

promedio_B = sum(B)/tamano

C = []

for i in range(tamano):
    C.append(A[i])
    C.append(B[i])

maximo = max(C)

indice_max = C.index(maximo)


print("Resultados")
print("Contenido del arreglo A:", A)
print("Contenido del arreglo B:", B)
print("Contenido del arreglo A + B:", suma_AyB)
print("Contenido del arreglo A * B:", producto_AyB)
print("Suma enteros pares A:", pares_enterosA)
print("Promedio del arreglo B:", promedio_B)
print("Contenido del arreglo C:", C)
print(f"El valor máximo del arreglo C es: {maximo} el cual está en el índice {indice_max}")

File: LAB1/laboratorio1_poo.cpp

/**
*  Laboratorio 1 POO python
*  Por: Jerónimo Hoyos Botero
*/

#include <iostream> 
#include <algorithm>
#include <cstdlib>
#include <ctime>
using namespace std;

void print_array(int* array, int n){
    for (int i = 0; i < n; i++){
        cout << array[i] << " ";
    }
}

int encontrar_maximo(int* arr, int n) {
    int max = arr[0];
    for (int i = 1; i < n; ++i) {
        if (arr[i] > max)
            max = arr[i];
    }
    return max;
}

int main(){
    int tamano;

    while (true) {
        cout << "Ingrese tamaño de lista menor o igual a 12: ";
        cin >> tamano;
        if (0 < tamano && tamano <= 12){break;};
    };

    int A[tamano],B[tamano],C[tamano*2];
    int valor,sumaAyB[tamano],productoAyB[tamano],pares_enterosA,sumaB,max,indice_max;

    for (int i = 0; i < tamano; i++){
        cout << "Ingrese valor entero para el índice " << i << " de A: ";
        cin >> valor;
        A[i] = valor;
    }
    
    srand(time(0));
    for (int i = 0; i < tamano; i++) {

        B[i] = rand() % 20 + 1; 
    }

    for (int i = 0; i < tamano; i++) {sumaAyB[i] = A[i] + B[i]; };

    for (int i = 0; i < tamano; i++) {productoAyB[i] = A[i] * B[i];};

    for (int i = 0; i < tamano; i++) {if(A[i]%2 == 0){pares_enterosA += A[i];}};

    sumaB=0;
    for (int i = 0; i < tamano; i++) {sumaB += B[i];};

    float promedioB = sumaB / tamano;

    for (int i = 0; i < tamano; i++) {
        C[i*2] = A[i];
        C[i*2+1] = B[i];
    };

    max = encontrar_maximo(C,tamano*2);

    for (int i = 0; i < tamano*2; i++) {
        if(max==C[i]){indice_max = i;}
    };
    
    cout << "Resultados";
    cout << "\n";
    cout << "Contenido del arreglo A: ";
    print_array(A,tamano);
    cout << "\n";
    cout << "Contenido del arreglo B: ";
    print_array(B,tamano);
    cout << "\n";
    cout << "Contenido del arreglo A + B: ";
    print_array(sumaAyB,tamano);
    cout << "\n";
    cout << "Contenido del arreglo A * B: ";
    print_array(productoAyB,tamano);
    cout << "\n";
    cout << "Suma enteros pares A: " << pares_enterosA;
    cout << "\n";
    cout << "Promedio del arreglo B: "<< promedioB;
    cout << "\n";
    cout << "Contenido del arreglo C: ";
    print_array(C,tamano*2);
    cout << "\n";
    cout << "El valor maximo del arreglo C es: "<< max << " el cual esta en el indice " << indice_max;

}